[Feature] Add script converts yolo-style to coco format (#161)
open-mmlab/mmyolo: docs/en/user_guides/useful_tools.md

```shell
python tools/dataset_converters/balloon2coco.py
```

- `yolo2coco.py` converts a dataset from `yolo-style` **.txt** format to COCO format, please use it as follows:

```shell
python tools/dataset_converters/yolo2coco.py --image-dir /path/to/the/root/dir/of/your_dataset [--split]
```

Instructions:

1. `-image-dir` is the root directory of the yolo-style dataset, which should contain `images`, `labels`, and `class.txt`. `class.txt` is the class declaration corresponding to the current dataset. One class a line.
2. `-split` indicates whether you need to format the dataset based on the existing `train`, `val`, and `text` categories. It is not used by default. If you need to use it, please ensure the corresponding `train.txt`, ` val.txt`, and `test.txt` must exist under `-image-dir`. Otherwise, the script will fail to run.
3. By default, the script will create a folder called `coco_format` in the `-image-dir` directory where stores the converted JSON file. If `-split` is not specified, the output file is `result.json`. If `-split` is specified, three corresponding files called `train.json`, `val.json`, and `test.json` will be generated.

> ````

Reviewer: ```suggestion
```

                                               --not-show
```

## Convert Dataset

The folder `tools/data_converters` currently contains `ballon2coco.py` and `yolo2coco.py` two dataset conversion tools.

- `ballon2coco.py` converts the `balloon` dataset (this small dataset is for starters only) to COCO format.

For a detailed description of this script, please see the `Dataset Preparation` section in [YOLOv5 from entry to deployment](./yolov5_tutorial.md).

```shell
python tools/dataset_converters/balloon2coco.py
```

- `yolo2coco.py` converts a dataset from `yolo-style` **.txt** format to COCO format, please use it as follows:

```shell
python tools/dataset_converters/yolo2coco.py --image-dir /path/to/the/root/dir/of/your_dataset [--split]
```

Instructions:

> 1. `-image-dir` is the root directory of the yolo-style dataset, which should contain `images`, `labels`, and `class.txt`. `class.txt` is the class declaration corresponding to the current dataset. One class a line as this example shows:

Reviewer: ```suggestion
1. `image-dir` is the root directory of the yolo-style dataset, which should contain `images`, `labels`, and `class.txt`. `class.txt` is the class declaration corresponding to the current dataset. One class a line as this example shows:
```

> >
> 
> ```
> |____ $ROOT_PATH
>     |__ class.txt
>     |__ labels
>     |__ images
>     |__ ...

Reviewer: ```suggestion
```bash
.
└── $ROOT_PATH
         ├── class.txt
         ├── labels
         ├── images
         └── ...
```

```

> 2. `-split` indicates whether you need to format the dataset based on the existing `train`, `val`, and `text` categories. It is not used by default. If you need to use it, please ensure the corresponding `train.txt`, ` val.txt`, and `test.txt` must exist under `-image-dir`. Otherwise, the script will fail to run.
> 3. By default, the script will create a folder called `coco_format` in the `-image-dir` directory where stores the converted JSON file. If `-split` is not specified, the output file is `result.json`. If `-split` is specified, three corresponding files called `train.json`, `val.json`, and `test.json` will be generated.

Reviewer: ```suggestion
2. `split` indicates whether you need to format the dataset based on the existing `train`, `val`, and `text` categories. It is not used by default. If you need to use it, please ensure the corresponding `train.txt`, ` val.txt`, and `test.txt` must exist under `image-dir`. Otherwise, the script will fail to run.
3. By default, the script will create a folder called `coco_format` in the `image-dir` directory where stores the converted JSON file. If `split` is not specified, the output file is `result.json`. If `split` is specified, three corresponding files called `train.json`, `val.json`, and `test.json` will be generated.
```

                                               --not-show
```

## Convert Dataset

The folder `tools/data_converters` currently contains `ballon2coco.py` and `yolo2coco.py` two dataset conversion tools.

- `ballon2coco.py` converts the `balloon` dataset (this small dataset is for starters only) to COCO format.

For a detailed description of this script, please see the `Dataset Preparation` section in [YOLOv5 from entry to deployment](./yolov5_tutorial.md).

```shell
python tools/dataset_converters/balloon2coco.py
```

- `yolo2coco.py` converts a dataset from `yolo-style` **.txt** format to COCO format, please use it as follows:

```shell
python tools/dataset_converters/yolo2coco.py --image-dir /path/to/the/root/dir/of/your_dataset [--split]
```

Instructions:

1. `image-dir` is the root directory of the yolo-style dataset, which should contain `images`, `labels`, and `class.txt`. `class.txt` is the class declaration corresponding to the current dataset. One class a line. The structure of the root directory should be formatted as this example shows:

```bash

> .
> └── $ROOT_PATH
>          ├── class.txt
>          ├── labels
>              ├── a.txt
>              ├── b.txt
>          ├── images
>              ├── a.jpg
>              ├── b.png
>          └── ...

Reviewer: ```suggestion
.
└── $ROOT_PATH
    ├── class.txt
    ├── labels
    │    ├── a.txt
    │    ├── b.txt
    │    └── ...
    ├── images
    │    ├── a.jpg
    │    ├── b.png
    │    └── ...
    └── ...
```

Reviewer: 其他也请改一下 😄

                                               --not-show
```

## Convert Dataset

The folder `tools/data_converters` currently contains `ballon2coco.py` and `yolo2coco.py` two dataset conversion tools.

- `ballon2coco.py` converts the `balloon` dataset (this small dataset is for starters only) to COCO format.

For a detailed description of this script, please see the `Dataset Preparation` section in [YOLOv5 from entry to deployment](./yolov5_tutorial.md).

```shell
python tools/dataset_converters/balloon2coco.py
```

- `yolo2coco.py` converts a dataset from `yolo-style` **.txt** format to COCO format, please use it as follows:

```shell
python tools/dataset_converters/yolo2coco.py --image-dir /path/to/the/root/dir/of/your_dataset [--split]
```

Instructions:

1. `image-dir` is the root directory of the yolo-style dataset, which should contain `images`, `labels`, and `class.txt`. `class.txt` is the class declaration corresponding to the current dataset. One class a line. The structure of the root directory should be formatted as this example shows:

```bash
.
└── $ROOT_PATH

>     ├── class.txt

Reviewer: 代码和这里的实例判断的文件要一致，代码中判断的是 `classes.txt`

Author: > 代码和这里的实例判断的文件要一致，代码中判断的是 `classes.txt`

Got it.

                                               --not-show
```

## Convert Dataset

The folder `tools/data_converters` currently contains `ballon2coco.py` and `yolo2coco.py` two dataset conversion tools.

- `ballon2coco.py` converts the `balloon` dataset (this small dataset is for starters only) to COCO format.

> For a detailed description of this script, please see the `Dataset Preparation` section in [YOLOv5 from entry to deployment](./yolov5_tutorial.md).

Reviewer 2: ```suggestion
For a detailed description of this script, please see the `Dataset Preparation` section in [From getting started to deployment with YOLOv5](./yolov5_tutorial.md).
```

```shell
python tools/dataset_converters/balloon2coco.py
```

- `yolo2coco.py` converts a dataset from `yolo-style` **.txt** format to COCO format, please use it as follows:

```shell

> python tools/dataset_converters/yolo2coco.py --image-dir /path/to/the/root/dir/of/your_dataset

Reviewer 2: ```suggestion
python tools/dataset_converters/yolo2coco.py  /path/to/the/root/dir/of/your_dataset
```

```

Instructions:

> 1. `image-dir` is the root directory of the yolo-style dataset, which should contain `images`, `labels`, and `classes.txt`. `classes.txt` is the class declaration corresponding to the current dataset. One class a line. The structure of the root directory should be formatted as this example shows:

Reviewer 2: ```suggestion
1. `image_dir` is the root directory of the yolo-style dataset, which should contain `images`, `labels`, and `classes.txt`. `classes.txt` is the class declaration corresponding to the current dataset. One class a line. The structure of the root directory should be formatted as this example shows:
```

```bash
.
└── $ROOT_PATH
    ├── classes.txt
    ├── labels
    │    ├── a.txt
    │    ├── b.txt
    │    └── ...
    ├── images
    │    ├── a.jpg
    │    ├── b.png
    │    └── ...
    └── ...
```

> 2. The script will automatically check if `train.txt`, `val.txt`, and `test.txt` have already existed under `image-dir`. If these files are located, the script will organize the dataset accordingly. Otherwise, the script will convert the dataset into one file. The image paths in these files must be **ABSOLUTE** paths.

Reviewer 2: ```suggestion
2. The script will automatically check if `train.txt`, `val.txt`, and `test.txt` have already existed under `image_dir`. If these files are located, the script will organize the dataset accordingly. Otherwise, the script will convert the dataset into one file. The image paths in these files must be **ABSOLUTE** paths.
```

> 3. By default, the script will create a folder called `annotations` in the `image-dir` directory which stores the converted JSON file. If `train.txt`, `val.txt`, and `test.txt` are not found, the output file is `result.json`. Otherwise, the corresponding JSON file will be generated, named as `train.json`, `val.json`, and `test.json`. The `annotations` folder may look similar to this:

Reviewer 2: ```suggestion
3. By default, the script will create a folder called `annotations` in the `image_dir` directory which stores the converted JSON file. If `train.txt`, `val.txt`, and `test.txt` are not found, the output file is `result.json`. Otherwise, the corresponding JSON file will be generated, named as `train.json`, `val.json`, and `test.json`. The `annotations` folder may look similar to this:
```

```bash
.
└── $ROOT_PATH
    ├── classes.txt
    ├── labels
    │    ├── a.txt
    │    ├── b.txt
    │    └── ...
    ├── images
    │    ├── a.jpg
    │    ├── b.png
    │    └── ...
    └── ...
```

2. The script will automatically check if `train.txt`, `val.txt`, and `test.txt` have already existed under `image-dir`. If these files are located, the script will organize the dataset accordingly. Otherwise, the script will convert the dataset into one file. The image paths in these files must be **ABSOLUTE** paths.
3. By default, the script will create a folder called `annotations` in the `image-dir` directory which stores the converted JSON file. If `train.txt`, `val.txt`, and `test.txt` are not found, the output file is `result.json`. Otherwise, the converted JSON file and their corresponding folders will be generated, named `train.json`, `val.json`, `test.json`, `train`, `val`, and `test`. `annotations` may look similar to this:

```bash
.
└── $ROOT_PATH
    ├── annotations
    │    ├── train.json

>     │    ├── train

Reviewer 2: Incorrect

python tools/misc/download_dataset.py --dataset-name voc2007
python tools/misc/download_dataset.py --dataset-name lvis
python tools/misc/download_dataset.py --dataset-name balloon [--save-dir ${SAVE_DIR}] [--unzip]

> ````

Reviewer: ```suggestion
```
```

Author: > three ` not four

These typos have all been fixed locally. Will push later.
